src/plugins/WebOfTrust/util/TickerDelayedBackgroundJob.java

```diff
@@ -0,0 +1,370 @@
+package plugins.WebOfTrust.util;
+
+import freenet.node.FastRunnable;
+import freenet.node.PrioRunnable;
+import freenet.support.Executor;
+import freenet.support.Ticker;
+import freenet.support.io.NativeThread;
+
+/**
+ * A {@link DelayedBackgroundJob} implementation that uses a {@link Ticker} for scheduling.
+ *
+ * @author bertm
+ * @see TickerDelayedBackgroundJobFactory
+ */
+public class TickerDelayedBackgroundJob implements DelayedBackgroundJob {
+    static enum JobState {
+        /** Waiting for a trigger, no running job thread or scheduled job. */
+        IDLE,
+        /** Waiting for the scheduled job to be executed, no running job thread. */
+        WAITING,
+        /** Running the job in the job thread. */
+        RUNNING,
+        /** Waiting for the running job thread to finish before we terminate. */
+        TERMINATING,
+        /** Terminated, no running job thread. */
+        TERMINATED
+    }
```

Comment:
I've added the above suggestion as a TODO to the code at commit bdc9b1e58ec589ec0a9c48979d07c98589cb945d